### PR TITLE
Possibility to document status code ranges in dynamic statuc code outputs

### LINKED ISSUE #1475: Support range definitions for response error status
It would be nice if response status [range definitions](https://swagger.io/docs/specification/describing-responses/) such as 4XX could be expressed with Tapir.

[This issue](https://github.com/softwaremill/tapir/issues/796) discusses handling status code ranges, but the suggested solution/workaround will result in rendering only one single status code to the OpenAPI doc.

### PR DESCRIPTION
**Pull request by adamw**
Closes #1475

### CODE REVIEW

File: core/src/main/scala/sttp/tapir/model/StatusCodeRange.scala [hunk 1]
```diff
@@ -0,0 +1,10 @@
+package sttp.tapir.model
+
+sealed abstract class StatusCodeRange(val range: Int)
+object StatusCodeRange {
+  case object Informational extends StatusCodeRange(1)
+  case object Success extends StatusCodeRange(1)
```
Comment (markarasev) [inline core/src/main/scala/sttp/tapir/model/StatusCodeRange.scala:6]: Shouldn't it be `StatusCodeRange(2)` here?

Comment (adamw) [reply core/src/main/scala/sttp/tapir/model/StatusCodeRange.scala:6]: of course 🤦 can you maybe create a PR with a fix?

Comment (markarasev) [reply core/src/main/scala/sttp/tapir/model/StatusCodeRange.scala:6]: https://github.com/softwaremill/tapir/pull/1924 but I'm getting out of memory errors in CI

Comment (adamw) [reply core/src/main/scala/sttp/tapir/model/StatusCodeRange.scala:6]: yeah the dotty compiler sometimes does that, no idea why

**Pull request merged**